src/runtime/ranger/ranger_resource_policy.h

```diff
@@ -0,0 +1,112 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <set>
+#include <vector>
+
+#include <rapidjson/document.h>
+
+#include "common/json_helper.h"
+#include "utils/fmt_logging.h"
+
+namespace dsn {
+namespace ranger {
+
+// ACL type defined in Range service for RPC matching policy
+enum access_type
+{
+    READ = 0,
+    WRITE,
+    CREATE,
+    DROP,
+    LIST,
+    METADATA,
+    CONTROL,
+    ALL,
+    INVALID,
+};
+
+ENUM_BEGIN(access_type, INVALID)
+ENUM_REG(READ)
+ENUM_REG(WRITE)
+ENUM_REG(CREATE)
+ENUM_REG(DROP)
+ENUM_REG(LIST)
+ENUM_REG(METADATA)
+ENUM_REG(CONTROL)
+ENUM_REG(ALL)
+ENUM_END(access_type)
+
+ENUM_TYPE_SERIALIZATION(access_type, INVALID)
+
+// Ranger policy data structure
+struct policy_item
+{
+    std::set<access_type> access_types;
```

Comment:
Use bitmap instead of std::set, `match` is a very frequently called function in hot path, we must ensure it's performance.
It's recommend to add a simple benchmark test for `match`.